### ISSUE TITLE
Use of placement_strategy within aws_ecs_service always forces new resource

### ISSUE THREAD

**Issue by fillup** [hashicorp/terraform#13216]
### Description
When using `aws_ecs_service`, if I specify a placement strategy, it forces a new aws_ecs_service resource on every apply. Even if the values do not change. This is problematic as deleting the existing service to recreate causes service disruption and create_before_destroy has issues because you can't create a second service with the same name.

### Terraform Version
Terraform v0.9.1

### Affected Resource(s)
- aws_ecs_service

### Terraform Configuration Files
```hcl
 resource "aws_ecs_service" "service" {
   name = "${var.service_name}"
   cluster = "${var.cluster_id}"
   desired_count = "${var.desired_count}"
   iam_role = "${aws_iam_role.ecsServiceRole.arn}"
   depends_on = ["aws_iam_role_policy.ecsServiceRolePolicy", "aws_alb_listener.https"]

   placement_strategy {
     type = "spread"
     field = "instanceId"
   }

   load_balancer {
     target_group_arn = "${aws_alb_target_group.tg.arn}"
     container_name = "${var.lb_container_name}"
     container_port = "${var.lb_container_port}"
   }

   # Track the latest ACTIVE revision
   task_definition = "${aws_ecs_task_definition.td.family}:${max("${aws_ecs_task_definition.td.revision}", "${data.aws_ecs_task_definition.td.revision}")}"
 }
```

### Debug Output
https://gist.github.com/fillup/af06c64f06943dfda6621898c866d48a

### Expected Behavior
I would expect no changes to occur since the setting for placement_strategy did not change. 

### Actual Behavior
The `aws_ecs_service` is destroyed and recreated. 

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. `terraform plan` - to see plan for replacement even though no changes to configuration were made
3. `terraform apply` - to see aws_ecs_service destroyed and recreated

### Important Factoids
I assume this is caused because `aws_ecs_service` can accept up to five `placement_strategy` blocks and so during plan/apply it doesn't recognize it is still the same one, but I'm just guessing.

**Comment by stack72** [comment]
Hi @fillup 

Thanks for the issue here - I can see the issue is actually down to casing :(

```
placement_strategy.#:                     "1" => "1"
placement_strategy.1676812570.field:      "instanceid" => "" (forces new resource)
placement_strategy.1676812570.type:       "spread" => "" (forces new resource)
placement_strategy.3946258308.field:      "" => "instanceId" (forces new resource)
placement_strategy.3946258308.type:       "" => "spread" (forces new resource)
```

This should not be an issue for the user so I am going to work on getting a fix together for 0.9.3

Sorry for this

Paul

**Comment by fillup** [comment]
Wow, that was amazingly fast, thank you!
